src/features/SignAndVerifyMessage/SignMessage.tsx

```diff
@@ -194,36 +143,52 @@ function SignMessage(props: Props) {
             label={translate('MSG_MESSAGE')}
             placeholder={translateRaw('SIGN_MSG_PLACEHOLDER')}
             textarea={true}
-            onChange={(event) => handleOnChange(event.target.value)}
+            onChange={(event) => messageUpdate(event.target.value)}
             height="150px"
-            inputError={error}
+            inputError={error && translateRaw('SIGN_MESSAGE_ERROR')}
           />
           <SignButton
             disabled={!message || isDemoMode}
             onClick={handleSignMessage}
-            loading={signStatus === SignStatus.SIGNING}
+            loading={status === 'SIGN_REQUEST'}
           >
-            {signStatus === SignStatus.SIGNING ? translate('SUBMITTING') : translate('NAV_SIGNMSG')}
+            {status === 'SIGN_REQUEST' ? translate('SUBMITTING') : translate('NAV_SIGNMSG')}
           </SignButton>
-          {signStatus === SignStatus.SIGNED && (
-            <SignedMessage>
-              <SignedMessageLabel>{translate('MSG_SIGNATURE')}</SignedMessageLabel>
-              <CodeBlockWrapper>
+          {status === 'SIGN_SUCCESS' && (
+            <Box mt="10px" width="100%">
+              <Text variant="label">{translate('MSG_SIGNATURE')}</Text>
+              <Box width="100%">
                 <CodeBlock>{JSON.stringify(signedMessage, null, 2)}</CodeBlock>
-              </CodeBlockWrapper>
-            </SignedMessage>
+              </Box>
+            </Box>
           )}
         </>
       )}
-    </Content>
+    </Box>
   );
 }
 
 const mapStateToProps = (state: AppState) => ({
-  isDemoMode: getIsDemoMode(state)
+  isDemoMode: getIsDemoMode(state),
+  status: selectSignMessageStatus(state),
+  error: selectSignMessageError(state),
+  signedMessage: selectSignedMessage(state),
+  message: selectMessage(state),
+  walletId: selectWalletId(state)
 });
 
-const connector = connect(mapStateToProps);
-type Props = ConnectedProps<typeof connector> & SignProps;
+const mapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
+  bindActionCreators(
+    {
+      walletSelect,
+      messageUpdate,
+      signMessageReset,
+      signMessage
+    },
+    dispatch
+  );
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+type Props = ConnectedProps<typeof connector> & OwnProps;
```

Comment:
Maybe use hooks instead of legacy connectors.

Reply:
For the moment the convention in core is to use `connect`. As all conventions it's open for a team discussion.

Reply:
We use hooks in a bunch of places, so I don't see why we can't do it here. They're easier to read and use, and should be very easy to replace here.